### PR TITLE
[SW-1636] Enable to specify outputCols on H2OTargetEncoder

### PR DESCRIPTION
**Pull request by mn-mikke**

### CODE REVIEW

File: ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala [hunk 1]
```diff
@@ -74,6 +74,8 @@ class H2OTargetEncoder(override val uid: String)
 
   def setInputCols(values: Array[String]): this.type = set(inputCols, values)
 
+  def setOutputCols(values: Array[String]): this.type = set(outputCols, values)
```
Comment (jakubhava) [inline ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala]: Should we check that `values` can't be null?

File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala [hunk 2]
```diff
@@ -75,7 +77,14 @@ trait H2OTargetEncoderParams extends Params {
 
   def getInputCols(): Array[String] = $(inputCols)
 
-  def getOutputCols(): Array[String] = getInputCols().map(_ + "_te")
+  def getOutputCols(): Array[String] = {
+    val columns = $(outputCols)
+    if (columns == null || columns.isEmpty) {
```
Comment (jakubhava) [inline scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala]: `outputCols` is `StringArrayParam` and it is not nullable, `null` should not be a valid option. I think we might just prevent user setting null, what do you thing?

Comment (jakubhava) [reply scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala]: Or we can keep the option to specify `null` but the parameter needs to be changed to `NullableStringArrayParam`

File: ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala (hunk 1 above)
Comment (mn-mikke) [reply ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala]: That's a really good point! In Scala world, it's not a practice, but an user can pass anything. Do check this in other parts of API? I don't have any strong opinion about this, I will tailor it according to you preferences ;-)

File: scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala (hunk 2 above)
Comment (mn-mikke) [reply scoring/src/main/scala/ai/h2o/sparkling/ml/params/H2OTargetEncoderParams.scala]: Yes, you are right, `null` is not a valid option. `columns == null` is extra

File: ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala (hunk 1 above)
Comment (jakubhava) [reply ml/src/main/scala/ai/h2o/sparkling/ml/features/H2OTargetEncoder.scala]: After today's discussion I would stick with the current API :)

**Pull request merged**